sklearn/model_selection/_split.py

```diff
@@ -440,13 +440,16 @@ class GroupKFold(_BaseKFold):
     The same group will not appear in two different folds (the number of
     distinct groups has to be at least equal to the number of folds).
 
-    The folds are approximately balanced in the sense that the number of
-    distinct groups is approximately the same in each fold.
-
     Parameters
     ----------
     n_splits : int, default=3
         Number of folds. Must be at least 2.
+    method: string, default='balance'
+        One of 'balance', 'stratify', 'shuffle'.
+        By default, try to equalize the sizes of the resulting folds.
+        If 'stratify', sort groups according to ``y`` variable and distribute
```

Comment:
It works best when there are many groups (i.e., groups are small), but it's not necessary to have a many-to-one relationship with the target variable. I now added an example where the target variable is a continuous variable. The plots show 1000 normally distributed data points, in 100 groups.

![cmp](https://user-images.githubusercontent.com/261460/28417564-dd3d890c-6d58-11e7-82f0-f9dac12ac834.png)